src/commands/insertFiles.ts

```diff
@@ -61,11 +61,15 @@ const insertFiles = function(view, event, pos, files, options) {
 
         // otherwise, insert it at the placeholder's position, and remove
         // the placeholder itself
-        const transaction = view.state.tr
-          .replaceWith(pos, pos, schema.nodes.image.create({ src }))
-          .setMeta(uploadPlaceholderPlugin, { remove: { id } });
-
-        view.dispatch(transaction);
+        const newImg = new Image();
+        newImg.onload = () => {
```

Comment:
This is a great improvement, It's been subconsciously bugging me for a while! 

We should add an `onerror` handler here that throws the err for the `catch` above, right?

Reply:
Good point. Done.

https://github.com/outline/rich-markdown-editor/pull/442/commits/728fa7470d38195a8a15aad768e6481c08ec1915